N24DCCN167/BaiTap_buoi8/7.cpp

```diff
@@ -16,20 +17,24 @@ int main()
         {
             if (newStr != "")
             {
-                words.push_back(newStr);
+                res.push_back(newStr);
                 newStr = "";
             }
         }
     }
     if (str[str.length() - 1] != ' ')
-        words.push_back(newStr);
-    string res = "";
-    for (int i = 0; i < words.size(); i++)
+        res.push_back(newStr);
+    cout << "[";
+    for (int i = 0; i < res.size(); i++)
     {
-        res = res + words[i];
-        if (i != words.size() - 1)
-            res += ' ';
+        if (i == res.size() - 1)
+        {
+            cout << '"' << res[i] << '"';
+            break;
+        }
+        cout << '"' << res[i] << '"' << ",";
     }
-    cout << "'" << res << "'";
+    cout << "]";
```

Comment:
Em mới chỉ update lại cách in thôi. Vẫn còn sử dụng mảng.
Đây là khung code của ý tưởng đúng
int main(){
xóa_dấu_cách_đầu_câu()
xóa_dấu_cách_cuối_câu()
// xóa dấu cách thừa ở giữa
while (tìm_kiếm_vị_trí_của_2_dấu_space_liên_tiếp != std::string::npos){
xóa_1_space_tại_vị_trí_đó()
}
// đoạn phía trên có nghĩa là: Khi nào còn tìm dc 2 dấu cách ở kề nhau (tức là có dấu cách dư) thì xóa 1 dấu đi. lặp lại cho tới khi nào xóa hết dấu cách
}